### ISSUE TITLE
No session manager found for affiliated R runtimes

### ISSUE THREAD

**Issue by juliasilge** [posit-dev/positron#5615]
#### Positron and OS details:

On current main, but I believe starting with https://github.com/posit-dev/positron/pull/5380

## Describe the issue:

In both release builds and dev builds, I cannot start up R or JavaScript runtimes. I _can_ start Zed and Python runtimes.

## Steps to reproduce the issue:

1. Start up Positron and notice that an associated R runtime does not start
2. Try to start a runtime by clicking and notice that nothing happens:

![Image](https://github.com/user-attachments/assets/4a9d2341-ecec-4443-a65a-476b5c729e73)

3. Try to start the dev-only JavaScript runtime via the command "Connect Extension Host Runtime" and notice nothing happens:

![Image](https://github.com/user-attachments/assets/d163c422-2a0e-4ed6-afae-36807e91adeb)

## Expected or desired behavior:

R and JavaScript runtimes should start.

## Were there any error messages in the UI, Output panel, or Developer Tools console?

In the Window output channel, I see errors like:

> 2024-12-04 11:36:02.879 [error] No session manager found for runtime dcf7187ac9fa9e0e20e9cdf6a71740d1 (language: R name: R 4.4.2 version: 4.4.2) (1 managers registered).: Error: No session manager found for runtime dcf7187ac9fa9e0e20e9cdf6a71740d1 (language: R name: R 4.4.2 version: 4.4.2) (1 managers registered).
    at RuntimeSessionService.getManagerForRuntime (vscode-file://vscode-app/Users/juliasilge/Work/posit/positron/out/vs/workbench/services/runtimeSession/common/runtimeSession.js:776:19)
    at async RuntimeSessionService.doAutoStartRuntime (vscode-file://vscode-app/Users/juliasilge/Work/posit/positron/out/vs/workbench/services/runtimeSession/common/runtimeSession.js:605:36)
> 2024-12-04 11:36:02.879 [error] No session manager found for runtime 13C365D6-099A-43EC-934D-353ADEFD798F (language: Node.js name: Node.js v20.15.1 version: v20.15.1) (1 managers registered).: Error: No session manager found for runtime 13C365D6-099A-43EC-934D-353ADEFD798F (language: Node.js name: Node.js v20.15.1 version: v20.15.1) (1 managers registered).
    at RuntimeSessionService.getManagerForRuntime (vscode-file://vscode-app/Users/juliasilge/Work/posit/positron/out/vs/workbench/services/runtimeSession/common/runtimeSession.js:776:19)
    at async RuntimeSessionService.doAutoStartRuntime (vscode-file://vscode-app/Users/juliasilge/Work/posit/positron/out/vs/workbench/services/runtimeSession/common/runtimeSession.js:605:36)

**Comment by juliasilge** [comment]
I observe this problem with [2025.01.0-2](https://github.com/posit-dev/positron-builds/releases/tag/2025.01.0-2) and `main` as of 2b838115abb371266250ff4742d5dd6a9f12b317

**Comment by juliasilge** [comment]
@jennybc got out of this state by nuking `/Users/jenny/Library/Application Support/Positron/`.

I am quite confident now this is about _affiliated_ runtimes that already exist.

**Comment by juliasilge** [comment]
We should address this before pushing another release build out to users.

**Comment by juliasilge** [comment]
I have been working with 2024.12.0-96 for the past day or so and had a lot of affiliated runtimes in various workspaces.

This morning I confirmed that 2025.01.0-10 still got me into a broken state as described here, where I could not start any R runtime at _all_ in many workspaces; this includes workspaces with affiliated runtimes, including affiliated runtimes that I don't have installed anymore (i.e. I updated R). 

By contrast, 2025.01.0-26 which includes #5618 does work as I expect. Affiliated R runtimes that I have still do start up, and in workspaces that have an affiliated runtime that I don't have anymore (i.e. older R) I can start R successfully.

This looks good to me, and I'm ready to call this fixed as far as we can tell for now! 🙌
